### PR TITLE
fix: do not use extra_args for enforced DashTestFramework arguments

### PR DESCRIPTION
**Pull request by knst**
## Issue being fixed or feature implemented
DashTestFramework requires 2 arguments for dashd which are lost every dashd restart: `sporkkey` and `dip3params`.
Without this PR you need to pass this arguments manually every time when you restart dashd in functional tests.

## What was done?
Make dash specific args `sporkkey` and `dip3params` resilient for dashd restart

It makes workarounds such as [these](https://github.com/dashpay/dash/commit/c28b05c5ca1d2b85687fa5e481e32208e7d48adc) no more needed: 
```
self.restart_node(1, self.extra_args[1] + ["-checkaddrman=1"])
```

Also there's some cleanup for `wallet_mnemonicbits.py` to remove `usehd=1` which is not required, and it has been discovered during revising all node's restarts related code.

## How Has This Been Tested?
Removed workaround from `rpc_net.py` and run functional tests

## Breaking Changes
N/A

## Checklist:
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

File: test/functional/test_framework/test_framework.py [hunk 1]
```diff
@@ -1119,9 +1119,7 @@ def set_dash_test_params(self, num_nodes, masterodes_count, extra_args=None, evo
             extra_args = [[]] * num_nodes
         assert_equal(len(extra_args), num_nodes)
         self.extra_args = [copy.deepcopy(a) for a in extra_args]
-        self.extra_args[0] += ["-sporkkey=cP4EKFyJsHT39LDqgdcB43Y3YXjNyjb5Fuas1GQSeAtjnZWmZEQK"]
-        for i in range(0, num_nodes):
-            self.extra_args[i].append("-dip3params=2:2")
+        self.extra_args_from_options += ["-sporkkey=cP4EKFyJsHT39LDqgdcB43Y3YXjNyjb5Fuas1GQSeAtjnZWmZEQK", "-dip3params=2:2"]
```
Comment (UdjinM6) [inline test/functional/test_framework/test_framework.py:1122]: 6a161abbce4310fa9ffd1763da93342b49ecad27: `sporkkey` should be set on node0 only. Also, we probably should be using `dash.conf` for options that are meant to be persistent, pls check 32373548a4.

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: test/functional/test_framework/test_framework.py (hunk 1 above)
Comment (knst) [reply test/functional/test_framework/test_framework.py:1122]: Why exactly sporkkey can't be specified for all nodes? It's out of scope for all tests except "feature_sporks.py" to care about sporks functionally

Comment (UdjinM6) [reply test/functional/test_framework/test_framework.py:1122]: You _can_ do that. But you shouldn't :D

**Review by UdjinM6 — APPROVED** [review]
utACK 6d4a7827560792a2951a5600bb43ff2b75028038

**Review by PastaPastaPasta — APPROVED** [review]
utACK 6d4a7827560792a2951a5600bb43ff2b75028038

**Pull request merged**